app/react-native/src/preview/index.js

```diff
@@ -133,15 +135,22 @@ export default class Preview {
     channel.emit(Events.GET_CURRENT_STORY);
   }
 
-  _setInitialStory = () => {
-    const story = this._getInitialStory();
+  _setInitialStory = async (shouldPersistSelection: boolean) => {
+    let story = this._getInitialStory();
+
+    if (shouldPersistSelection) {
+      const value = await AsyncStorage.getItem(STORAGE_KEY);
```

Comment:
I suggest to also have some kind of check to make sure that we don't get garbage from AsyncStorage, I had few cases where I didn't get what I expected